pubpub/platform
# Allow creating a pub without a stage

## Issue(s) Resolved
Closes https://github.com/pubpub/platform/issues/557

## High-level Explanation of PR

Allows the stage to be undefined on pub create. There's a new dropdown in the stage dropdown that is "No stage" which leads to an undefined stage.

The bulk of this change was actually using `createPubRecursiveNew` instead of `createPub`. `createPubRecursiveNew` is used by the API and already takes an optional stage ID. I had to rejigger `createPubRecursiveNew` a little bit to allow for empty pub values

## Test Plan

* Visit http://localhost:3000/c/croccroc/pubs
* Click "Create"
* The create form should pop up with "No stage" as the default under "Stage"
* You should be able to save the pub without selecting a stage

## Screenshots (if applicable)
<img width="531" alt="image" src="https://github.com/user-attachments/assets/e23dd1bd-2dde-4783-a817-54a8373e0300">


## Notes
